diagridio/catalyst-quickstarts
# Quickstart script/code updates for public preview (WIP)

Testing/update progress (this means i have updated the scripts + code and tested the code and scripts)
- [x] Csharp 
- [x] Java (testing in progress)

Updates for run.py/scaffold.py across language: 
- Update the run.py file to write with spinner.write vs print to avoid print conflicts 
- Remove unnecessary/redundant errors from being printed
- Make app dependency checks more streamlined
- Add timer to app id status check command to avoid reprints of the same message + actionable message on failure 
- Update spinner behavior to update the action with a success or failure and only print out additional information if context/error needed to be displayed 
- Remove output related to venv due to lack of benefit to user / clarity on what is happening (we can add this as a disclaimer in docs or put a one liner to describe why we are doing that to update the dev config) 
- add variables where possible
- remove app ports from apps which do not need a dev tunnel

Update for invocation across languages: 
- Remove prefixing on rest APIs ex ( /pubsub/x, /invoke/y) as no longer needed with move to separate api projects
- Remove caller/target and request/reply references to standardize on client/server language as requested by yaron 
- Update dev config to align to respective app id name changes, make sure workdirs are aligned
- make sure the various samples return similar messages on success and failure to create consistency 

Update for state across languages: 
- Remove invocation prefixing on rest APIs as no longer needed with move to single api projects
- Update messages/env to reflect "state item" vs. key/value to align to state language
- Make sure the various samples return similar messages on success and failure to create consistency 

Update for workflow: 
- Update the app id name to be order-wf to reflect more context of the business scenario vs generic workflow-app

Areas with additional help needed: 
- I don't understand the case where an existing dev config with the same name would be present? Can someone help me understand the scenario here? 

Testing performed to validate: 
- Versioning scenarios (dotnet not present, lower version, etc.) were tested - still working through languages
- App ID provisioning failure tested 
- Project name conflict tested 
- Project max capacity tested 
- Subscription failure tested
